### PR TITLE
ACM-9905 Kubevirt Cluster creation - hosted cluster namespace combobox addition

### PR DESCRIPTION
**Pull request by vishsanghishetty**
## Description
This PR is for [ticket](https://issues.redhat.com/browse/ACM-9905) ACM-9905 Kubevirt Cluster creation - hosted cluster namespace combobox addition

## Acceptance Criteria
- [x] Adds Hosted cluster namespace field to Cluster Wizard for Kubevirt clusters
- [x] Unit tests updated with the combo box displaying `clusters` as the default value and user-defined hosted cluster namespace

image

image
image

### CODE REVIEW

**Comment by vishsanghishetty** [comment]
@rokej @fxiang1 

Kevin and I have the following questions regarding the **Hosted Cluster namespace** field in KubeVirt cluster creation:

1. Tooltip for this field, something like this:  
   "The namespace where hosted cluster resources will be created. Do not enter a namespace that matches the cluster's name or use an existing managed cluster namespace."
   - Is this correct?  
   - Should we include additional information to explain why this is important?  

2. Should we add validation to prevent the use of a managed cluster namespace, or might there be cases where the user should be allowed to do this for some reason?  

image

**Comment by fxiang1** [comment]
> @rokej @fxiang1
> 
> Kevin and I have the following questions regarding the **Hosted Cluster namespace** field in KubeVirt cluster creation:
> 
>     1. Tooltip for this field, something like this:
>        "The namespace where hosted cluster resources will be created. Do not enter a namespace that matches the cluster's name or use an existing managed cluster namespace."
>        
>        * Is this correct?
>        * Should we include additional information to explain why this is important?
> 
>     2. Should we add validation to prevent the use of a managed cluster namespace, or might there be cases where the user should be allowed to do this for some reason?

Thanks @vishsanghishetty 
For 1. this looks good. Yes, I think it will be helpful to include additional information to explain why this is important. Like saying the hosted cluster will be deleted during detach if the user does that.

For 2. I'm not sure if we can prevent it completely. Even when we prevent the user from using a managed cluster namespace on the form they can still use the YAML editor to change it.

**Comment by fxiang1** [comment]
BTW this fix https://issues.redhat.com/browse/ACM-15018 from the server foundation team will prevent namespaces with hosted clusters from deletion during a detach.

**Comment by rokej** [comment]
In my opinion, it is not a recommended practice to create resources in a managed cluster namespace. Therefore, if there is a chance to prevent it, we should prevent it even though the backend has implemented with https://issues.redhat.com/browse/ACM-15018, it will not be apparent to users why the namespace is not deleted when a managed cluster is detached.

**Comment by KevinFCormier** [comment]
@vishsanghishetty Can you please add the tooltip text and validation?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
Works well, but some code cleanup needed. Please see inline comments. Thanks.

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx [hunk 1]
```diff
@@ -287,13 +288,13 @@ export default function CreateCluster(props: { infrastructureType: ClusterInfras
...
-          const namespace = cluster?.metadata?.namespace
-          if (!noRedirect && clusterName && namespace) {
+          const namespace = cluster?.metadata?.namespace ?? null
+          if (!noRedirect && clusterName && clusterNamespace) {
             setTimeout(() => {
               navigate(
                 generatePath(NavigationPath.clusterDetails, {
                   name: clusterName,
-                  namespace,
+                  namespace: namespace,
```
[the remark this answers is not in the document]
Comment (vishsanghishetty) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx:297]: forgot to clean this up, thanks for pointing this

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/controlData/ControlDataKubeVirt.js [hunk 2]
```diff
@@ -111,6 +115,25 @@ export const getControlDataKubeVirt = (
...
+        constraint: VALID_DNS_LABEL,
+        notification: t('import.form.invalid.dns.label'),
+        required: true,
+      },
+    },
+    {
+      id: 'userDefinedNamespace',
+      type: 'hidden',
+      active: '',
+    },
```
[the remark this answers is not in the document]
Comment (vishsanghishetty) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/controlData/ControlDataKubeVirt.js:136]: forgot to clean this up, thanks for pointing this

File: frontend/public/locales/en/translation.json [hunk 3]
```diff
@@ -3020,6 +3021,7 @@
   "tooltip.creation.ocp.diskSizeGB": "Disk size (GiB)",
   "tooltip.creation.ocp.gcp.instance.type": "The GCP machine type.",
   "tooltip.creation.ocp.gcp.region": "The GCP region where you deploy your cluster. You can select zones within the region for your control plane and worker pools.",
+  "tooltip.creation.ocp.hosted.cluster.namespace": "The namespace where hosted cluster resources will be created. Do not enter a namespace that matches the cluster's name or use an existing managed cluster namespace.",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:3024]:
```suggestion
  "tooltip.creation.ocp.hosted.cluster.namespace": "The namespace where hosted cluster resources are created. Do not enter a namespace that matches the cluster name and do not use an existing managed cluster namespace.",
```

I assume for the last part, it's supposed to be a negative for both?

Comment (KevinFCormier) [reply frontend/public/locales/en/translation.json:3024]: Yup, that's right. Great edits. Thank you!

**Pull request merged**